src/include/detail/linalg/tdb_matrix.h

```diff
@@ -351,20 +351,44 @@ class tdbBlockedMatrix : public MatrixBase {
 
     auto layout_order = schema_.cell_order();
 
-    // Create a query
+    // Read TileDB data
+    size_t read_batch_size_cells = get_read_batch_size_cells(ctx_);
+    size_t total_size = elements_to_load * dimension;
+    size_t offset = 0;
     tiledb::Query query(ctx_, *array_);
-    query.set_subarray(subarray)
-        .set_layout(layout_order)
-        .set_data_buffer(attr_name, this->data(), elements_to_load * dimension);
-    tiledb_helpers::submit_query(tdb_func__, uri_, query);
+    query.set_subarray(subarray).set_layout(layout_order);
+    tiledb::Query::Status status;
+    do {
```

Comment:
Does this result in incomplete queries whenever `read_batch_size_cells` is less than `total_size`?  I don't think we should batch read unless we actually need to.  That is, we should always try to read `total_size` and then iterate if we don't get `total_size`.  

Having incomplete queries here is a very edge case.  Reading in batches may or may not solve that issue, but it adds latency to the common case.

Reply:
One rule in software is "don't pay for what you don't need."  It is true that there is only a 5% penalty with this particular set of assumptions.  However, I expect the server is going to deal with its memory footprint in its own way, which is why we might get incomplete reads.  I would say just request what is needed and let the server say how much memory it wants to devote to that or not.

Do other apps try to be parsimonious this way?

I guess, on the other hand, since this is configurable, we can always just make it really large.